src/complex/Parameters/FileSystemPathParameter.cpp

```diff
@@ -166,9 +167,27 @@ Result<> FileSystemPathParameter::validatePath(const ValueType& path) const
       return {nonstd::make_unexpected(std::vector<Error>{{-2, "File System Path must include a file extension"}})};
     }
 
-    if(path.has_extension() && !m_AvailableExtensions.empty() && m_AvailableExtensions.find(path.extension().string()) == m_AvailableExtensions.end())
+    if(path.has_extension() && !m_AvailableExtensions.empty())
     {
-      return {nonstd::make_unexpected(std::vector<Error>{{-3, fmt::format("File extension '{}' is not a valid file extension", path.extension().string())}})};
+      bool validExtension = false;
+      for(const auto& ext : m_AvailableExtensions)
+      {
+        std::string inputExt = complex::StringUtilities::toLower(path.extension().string());
+        std::string compareExt = complex::StringUtilities::toLower(ext);
+        if(compareExt.at(0) == '.' && inputExt[0] != '.')
+        {
+          compareExt = complex::StringUtilities::chopr(compareExt, 1);
+        }
```

Comment:
`std::filesystem::path::extension()` always includes the `.` so instead we could force the available extension to have it in the constructor as well as making them lowercase to avoid doing it in every `validate` call. Then this loop can be:
```cpp
std::string inputExt = StringUtilities::toLower(path.extension().string());
bool validExtension = std::find(m_AvailableExtensions.cbegin(), m_AvailableExtensions.cend(), inputExt) != m_AvailableExtensions.cend()
```

Additionally while checking the documentation for this, I noticed that it behaves like `QFileInfo::suffix()` not `QFileInfo::completeSuffix()`. I don't know if this makes a difference in any code we have, but I thought it was important to mention just in case.